ring/partition_ring_desc.proto

```diff
@@ -32,16 +32,22 @@ message PartitionDesc {
 enum PartitionState {
   PartitionUnknown = 0;
 
+  // Pending partition is a partition that is about to be switched to ACTIVE. This state is used
+  // to let owners to attach to the partition and get ready to handle the partition.
+  //
+  // When a partition is in this state, it must not be used for writing or reading.
+  PartitionPending = 1;
```

Comment:
Note to reviewers: this will break (again) existing rings, but given the feature it's under development I would prefer to sort states by their expected sequence of events.

Reply:
:+1: Makes sense.